libs/dh/admin/shared/src/dh-permissions-table.component.ts

```diff
@@ -87,12 +83,11 @@ export class DhPermissionsTableComponent implements OnChanges {
     description: { accessor: 'description' },
   };
 
-  ngOnChanges() {
-    this.dataSource.data = this.permissions;
-    if (this.permissionsTable) this.permissionsTable.clearSelection();
-  }
+  ngOnChanges(): void {
+    this.dataSource.data = this.permissions();
 
-  onSelectionChange(selections: PermissionDetailsDto[]): void {
-    this.selectionChanged.emit(selections);
+    if (this.permissionsTable) {
```

Comment:
Issue: permissionTable will always be something, because its a signal

const table = this.permissionTable();
if(table) ...

Or maybe make its a viewChild.required

Reply:
I tried with `viewChild.required` but got a run-time error because the table couldn't be found. It's probably something related to `ngOnChanges` running too early. It's worth looking at this component again on its own.

Reply:
Ok, but dont you still need to 

const table = this.permissionTable();
if(table) ... or you are checking if the signal is there and not what the signal returns